StardewArchipelago/Items/Unlocks/UnlockManager.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Xna.Framework;
+using StardewArchipelago.Archipelago;
+using StardewArchipelago.Items.Mail;
+using StardewArchipelago.Locations.CodeInjections;
+using StardewValley;
+
+namespace StardewArchipelago.Items.Unlocks
+{
+    public class UnlockManager
+    {
+        private List<IUnlockManager> _specificUnlockManagers;
+
+        public UnlockManager(ArchipelagoClient archipelago)
+        {
+            _specificUnlockManagers = new List<IUnlockManager>();
+            _specificUnlockManagers.Add(new VanillaUnlockManager());
+            if (archipelago.SlotData.Mods.IsModded)
+            {
+                _specificUnlockManagers.Add(new ModSkillUnlockManager());
+                _specificUnlockManagers.Add(new MagicUnlockManager());

```

Comment:
I **really** like the initiaive of splitting this. Great!

But also.... maybe just add them if their mod is active? ;)